### PR TITLE
akka-persistence compiler warnings as fatal errors

### PR DESCRIPTION
**Pull request by johanandren**
Refs #26088

Scala/javadoc validation failing, waiting for #26634 to get that disabled

### CODE REVIEW

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala [hunk 1]
```diff
@@ -86,14 +86,6 @@ private[persistence] trait Eventsourced
   private val writerUuid = UUID.randomUUID.toString
 
   private var journalBatch = Vector.empty[PersistentEnvelope]
-  // no longer used, but kept for binary compatibility
```
Comment (chbatey) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: lies

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala [hunk 2]
```diff
@@ -249,8 +241,8 @@ private[persistence] trait Eventsourced
     require(persistenceId.trim.nonEmpty, s"persistenceId cannot be empty for PersistentActor [${self.path}]")
 
     // Fail fast on missing plugins.
-    val j = journal;
```
Comment (chbatey) [inline akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: :eyes: ;

File: akka-persistence/src/main/scala/akka/persistence/Persistence.scala [hunk 3]
```diff
@@ -404,7 +404,7 @@ class Persistence(val system: ExtendedActorSystem) extends Extension {
     log.debug(s"Create plugin: $pluginActorName $pluginClassName")
     val pluginClass = system.dynamicAccess.getClassFor[Any](pluginClassName).get
     val pluginDispatcherId = pluginConfig.getString("plugin-dispatcher")
-    val pluginActorArgs = try {
+    val pluginActorArgs: List[AnyRef] = try {
```
Comment (chbatey) [inline akka-persistence/src/main/scala/akka/persistence/Persistence.scala]: what was the warning here?

File: akka-persistence/src/main/scala/akka/persistence/serialization/MessageSerializer.scala [hunk 4]
```diff
@@ -46,12 +47,12 @@ class MessageSerializer(val system: ExtendedActorSystem) extends BaseSerializer
...
-    case a: AtomicWrite                 => atomicWriteBuilder(a).build().toByteArray
-    case a: AtLeastOnceDeliverySnapshot => atLeastOnceDeliverySnapshotBuilder(a).build.toByteArray
-    case s: StateChangeEvent            => stateChangeBuilder(s).build.toByteArray
-    case p: PersistentFSMSnapshot[Any]  => persistentFSMSnapshotBuilder(p).build.toByteArray
-    case _                              => throw new IllegalArgumentException(s"Can't serialize object of type ${o.getClass}")
+    case p: PersistentRepr                        => persistentMessageBuilder(p).build().toByteArray
+    case a: AtomicWrite                           => atomicWriteBuilder(a).build().toByteArray
+    case a: AtLeastOnceDeliverySnapshot           => atLeastOnceDeliverySnapshotBuilder(a).build.toByteArray
+    case s: StateChangeEvent                      => stateChangeBuilder(s).build.toByteArray
+    case p: PersistentFSMSnapshot[Any @unchecked] => persistentFSMSnapshotBuilder(p).build.toByteArray
```
Comment (chbatey) [inline akka-persistence/src/main/scala/akka/persistence/serialization/MessageSerializer.scala]: Can be [_]?

File: akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala [hunk 5]
```diff
@@ -705,7 +700,7 @@ object PersistentActorSpec {
       case Cmd(data) =>
         sender() ! data
         persist(()) { _ =>
-        } // skip calling defer immediately because of empty pending invocations
+          } // skip calling defer immediately because of empty pending invocations
```
Comment (chbatey) [inline akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala]: this really the format?

**Review by chbatey — APPROVED** [review]
LGTM

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala (hunk 1 above)
Comment (johanandren) [reply akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: Yeah, I can't see how a private field could affect bincomp?

File: akka-persistence/src/main/scala/akka/persistence/Persistence.scala (hunk 3 above)
Comment (johanandren) [reply akka-persistence/src/main/scala/akka/persistence/Persistence.scala]: Inferring the type to `List[Object]` ("is probably not what you want")

File: akka-persistence/src/main/scala/akka/persistence/serialization/MessageSerializer.scala (hunk 4 above)
Comment (johanandren) [reply akka-persistence/src/main/scala/akka/persistence/serialization/MessageSerializer.scala]: Actually, it couldn't:
```
[error] /Users/johan/Code/Akka/akka/akka-persistence/src/main/scala/akka/persistence/serialization/MessageSerializer.scala:54:73: type mismatch;
[error]  found   : akka.persistence.fsm.PersistentFSM.PersistentFSMSnapshot[_]
[error]  required: akka.persistence.fsm.PersistentFSM.PersistentFSMSnapshot[Any]
```

File: project/AkkaDisciplinePlugin.scala [hunk 6]
```diff
@@ -132,6 +108,8 @@ object AkkaDisciplinePlugin extends AutoPlugin with ScalafixSupport {
     "-Ywarn-unused:_",
     "-Ypartial-unification",
     "-Ywarn-extra-implicit",
-  )
+    "akka-protobuf",
```
Comment (raboof) [inline project/AkkaDisciplinePlugin.scala]: Huh? merge mistake?

Comment (johanandren) [reply project/AkkaDisciplinePlugin.scala]: Yeah, there was something that messed up merging this in all my PRs

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala (hunk 1 above)
Comment (patriknw) [reply akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: Example:
```
:javap -p akka.persistence.NamedPersistentActor

  private final int akka$persistence$Eventsourced$$maxMessageBatchSize;
  public int akka$persistence$Eventsourced$$maxMessageBatchSize();
  public final void akka$persistence$Eventsourced$_setter_$akka$persistence$Eventsourced$$maxMessageBatchSize_$eq(int);
```

Adding a field would not be allowed, removing might be ok, but do we dare to take the risk? Could we at least wait until 2.6?

**Review by patriknw — COMMENTED** [review]
looking good, but I think we should avoid risks by touching that dreadful trait

File: akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala (hunk 1 above)
Comment (johanandren) [reply akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: Alright, I'll revert that and see if I can silence it in some other way (I started out with but didn't get @unused to work)

Comment (chbatey) [reply akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: @silent should work

Comment (johanandren) [reply akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala]: reverted

**Review by raboof — APPROVED** [review]

**Pull request merged**